### PR TITLE
enable ci-tests and ci-wheels

### PR DESCRIPTION
**Pull request by bjlittle**
Apologies for this being a bit of a monster :japanese_ogre: 

Normally I'd split this up into several different PRs, but we're super keen to make a release of `python-stratify` and this PR ticks the remaining infra-structure boxes (hopefully)

Essentially, this PR:
- retires `cirrus-ci` and adopts github actions for CI
- updates the pattern for gha `on: branches`
- adds the `ci-tests` gha for testing
- adds the `ci-wheels` gha for building, testing and publishing `sdist` and binary `wheels`, providing support for `py39`, `py310` and `py311` on:
  - `linux` for `x86_64` (`manylinux2014` [PEP599](https://peps.python.org/pep-0599/))
  - `macosx` for `x86_64` and `arm64`
  - `win` for `amd64`
- updates the `MANIFEST.in` in alignment with the pattern adopted on `SciTools/iris`
- refreshes the `README.md`
- adds automated test coverage using [codecov](https://about.codecov.io/)
- enforces the [NEP29 Drop Schedule](https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule), thus raising the minimum supported version of `python` to be `>=3.9`, and `numpy` to be `>=1.21`
- adopts [PEP621](https://peps.python.org/pep-0621/) to retire our use of `setup.cfg` and migrate to `pyproject.toml`
- moves `stratify` codebase under the `src` directory
- modernises the `setup.py`, given [PEP518](https://peps.python.org/pep-0518/)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,38 +1,78 @@
...
 
+[project]
+authors = [
+    {name = "Stratify Contributors", email = "scitools.pub@gmail.com"}
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "License :: OSI Approved :: BSD License",
```
Comment (HGWright) [inline pyproject.toml:21]: repeat of line 20

**Review by HGWright — CHANGES_REQUESTED** [review]
Looks great and should be able to merge and release once a couple of minor changes have been made.

File: .github/workflows/ci-wheels.yml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      - "v*"
+
+jobs:
+  build_bdist:
+    name: "build ${{ matrix.os }} wheels"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        # TBD: extend to support windows
```
[the remark this answers is not in the document]
Comment (bjlittle) [reply .github/workflows/ci-wheels.yml]: See [788e044](https://github.com/SciTools/python-stratify/pull/79/commits/788e04469e7b533ca5d74b79a83b030a296bec81)

File: pyproject.toml (hunk 1 above)
Comment (bjlittle) [reply pyproject.toml]: See [788e044](https://github.com/SciTools/python-stratify/pull/79/commits/788e04469e7b533ca5d74b79a83b030a296bec81)

**Comment by bjlittle** [comment]
@HGWright Sweeeeet!

Nice one fella! Thanks :beers:

**Pull request merged**